x-pack/plugins/code/public/reducers/search.ts

```diff
@@ -45,7 +46,7 @@ const initialState: SearchState = {
   isLoading: false,
   isScopeSearchLoading: false,
   scope: SearchScope.DEFAULT,
-  searchOptions: { repoScope: [] },
+  searchOptions: { repoScope: [], defaultRepoScopeOn: true },
```

Comment:
we should not set `defaultRepoScopeOn` in the initialState`. let's trigger an action to set the value to be true when ther `repoScope` array is empty.